### PR TITLE
Taniaksiazka scraper refactor

### PR DESCRIPTION
**Pull request by piotr921**

### CODE REVIEW

File: czytamplscraper/src/main/java/com/blackbook/czytamplscraper/service/ScraperService.java [hunk 1]
```diff
@@ -24,15 +17,18 @@
     @Value("${endpoints.persistence-api}")
     private String persistenceApiEndpoint;
 
+    private Scraper scraper;
+    private RestOperations restOperations;
+
+    public ScraperService(Scraper scraper, RestOperations restOperations) {
```
Comment (mprtcz) [inline czytamplscraper/src/main/java/com/blackbook/czytamplscraper/service/ScraperService.java:23]: How is it autowired

File: czytamplscraper/src/test/java/com/blackbook/czytamplscraper/service/ScraperServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.*;
+
+public class ScraperServiceTest {
+
+    @Test
+    public void shouldSaveResultsInDatabase() throws Exception {
+        // Given
```
Comment (mprtcz) [inline czytamplscraper/src/test/java/com/blackbook/czytamplscraper/service/ScraperServiceTest.java:23]: SHould not add those comments by now

File: taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReader.java [hunk 3]
```diff
@@ -2,28 +2,50 @@
...
 
 @Slf4j
+@Component
 public class PromoDetailsReader {
     private static final int BOOKSTORE_ID = 3;
     private static final String BOOKSTORE_URL = "http://www.taniaksiazka.pl";
 
     BookDiscountData readDiscountDataProperties(Element book) {
-        log.info(book.select("a").attr("data-name"));
+        log.info(readTitle(book));
```
Comment (agawda) [inline taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReader.java:16]: :+1:  for extracting method.

File: taniaksiazkascraper/src/test/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReaderTest.java [hunk 4]
```diff
@@ -27,6 +28,27 @@ public void shouldReadPromotionDetails() throws IOException {
 
         // Then
         assertEquals("Cztery pory roku w afgańskiej wiosce", discountData.getBookData().getTitle());
+        assertEquals(null, discountData.getBookData().getSubtitle());
```
Comment (mprtcz) [inline taniaksiazkascraper/src/test/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReaderTest.java:31]: WHY NOT assertNull()?

**Review by agawda — APPROVED** [review]

**Pull request merged**